core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -454,6 +454,30 @@ public Table emptyCopy(int rowSize) {
         tables[1] = where(table2Selection);
         return tables;
     }
+    
+    /**
+     * Splits the table into two stratified samples, this uses the specified column to divide the table into groups, randomly assigning records to each according to the proportion given in trainingProportion. 
+     * 
+     *
+     * @param column the column to be used for the stratified sampling
+     * @param table1Proportion The proportion to go in the first table
+     * @return An array two tables, with the first table having the proportion specified in the method parameter,
+     * and the second table having the balance of the rows
+     */
+    public Table[] stratifiedSampleSplit(CategoricalColumn column, double table1Proportion){
+        Preconditions.checkArgument( CategoricalColumn.class.isInstance(this.column(column.name())),
```

Comment:
Why not do `isInstance(column)` instead of looking up the column we already have from the table?